service/src/main/java/greencity/service/HabitAssignServiceImpl.java

```diff
@@ -231,40 +230,27 @@ public Integer getReadinessPercent(HabitAssignDto habitAssign) {
     }
 
     /**
-     * {@inheritDoc}
+     * Method which updates duration of habit assigned for user.
+     *
+     * @param habitAssignId {@code AssignHabit} id.
+     * @param userId        {@link Long} id.
+     * @param duration      {@link Integer} with needed duration.
+     * @return {@link HabitAssignUserDurationDto}.
      */
     @Transactional
     @Override
-    public HabitAssignUserShoppingListItemDto updateUserShoppingItemListAndDuration(Long habitId, Long userId,
-        HabitAssignPropertiesDto habitAssignPropertiesDto) {
-        if (!habitRepo.existsById(habitId)) {
-            throw new NotFoundException(ErrorMessage.HABIT_NOT_FOUND_BY_ID + habitId);
+    public HabitAssignUserDurationDto updateUserHabitInfoDuration(Long habitAssignId, Long userId, Integer duration) {
+        if (!habitAssignRepo.existsById(habitAssignId)) {
+            throw new NotFoundException(ErrorMessage.HABIT_NOT_FOUND_BY_ID + habitAssignId);
         }
-        HabitAssign habitAssign = habitAssignRepo.findByHabitIdAndUserIdAndStatusIsInprogress(habitId, userId)
+        HabitAssign habitAssign = habitAssignRepo.findByHabitAssignIdUserIdAndStatusIsInProgress(habitAssignId, userId)
             .orElseThrow(() -> new InvalidStatusException(ErrorMessage.HABIT_ASSIGN_STATUS_IS_NOT_INPROGRESS));
```

Comment:
`habitAssignRepo.findByHabitAssignIdUserIdAndStatusIsInProgress` will also return `Optional.empty` for `INPROGRESS` habitAssign if ` ha.user.id != :userId`
See https://github.com/ita-social-projects/GreenCity/pull/5531/files#diff-05169418d0f08eeb9efc32c9ae364ca45acf5015e13946bd7fffc632a1546bdcR98
Should we use another exception message in this case?